### PR TITLE
[IMP] developer: rewrite the page on assets

### PR DESCRIPTION
**Pull request by ged-odoo**

### CODE REVIEW

File: content/developer/reference/frontend/assets.rst [hunk 1]
```diff
@@ -38,191 +42,193 @@ a dedicated `assets` key which contains a dictionary. The dictionary will map
...
+  website: ecommerce, forum, blog, event management, ...
 
-  - the `t-call-assets` directive with the `t-js` attribute set to false will
-    be replaced by a list of stylesheet tags pointing to the css files
+- `web.assets_qweb`: all static XML templates used in the backend environment
+  and in the point of sale.
 
-  - the `t-call-assets` directive with the `t-css` attribute set to false will
-    link to the non minified bundle file (which uses sourcemaps)
+- `web.qunit_suite_tests`: all javascript qunit testing code (tests, helpers, mocks)
```
Comment (aab-odoo) [inline content/developer/reference/frontend/assets.rst:62]: is it worth mentioning the qunit_mobile_suite_tests bundle?

File: content/developer/reference/frontend/assets.rst [hunk 2]
```diff
@@ -38,191 +42,193 @@ a dedicated `assets` key which contains a dictionary. The dictionary will map
...
 
-Here are some important bundles that most developers will need to know:
+    Whenever the browser loads odoo, it calls the `/web/webclient/qweb/` controller
+    to fetch the templates.
 
-- `web.assets_common`: this bundle contains most assets which are common to the
-  web client, the website, and also the point of sale. This is supposed to contain
-  lower level building blocks for the odoo framework. Note that it contains the
-  :file:`boot.js` file, which defines the odoo module system.
+It is useful to know that in most cases, a browser will actually only performs a
```
Comment (aab-odoo) [inline content/developer/reference/frontend/assets.rst:92]: ~~will~~

File: content/developer/reference/frontend/assets.rst [hunk 3]
```diff
@@ -38,191 +42,193 @@ a dedicated `assets` key which contains a dictionary. The dictionary will map
...
+    'web.assets_common': [
+        ('after', 'web/static/src/css/list_view.scss', 'my_addon/static/src/css/list_view.scss'),
+    ],
 
-- Use nested bundles: `include`
-    The `include` directive is a way to use a same bundle in other bundles to minimize
-    the size of your manifest. In Odoo we use sub bundles (prefixed with an underscore
-    by convention) to batch files used in multiple other bundles. You can then
-    specify the sub bundle as a pair `('include', <bundle>)` like this:
+`inlude`
```
Comment (aab-odoo) [inline content/developer/reference/frontend/assets.rst:159]: in**c**lude

File: content/developer/reference/frontend/assets.rst [hunk 4]
```diff
@@ -38,191 +42,193 @@ a dedicated `assets` key which contains a dictionary. The dictionary will map
...
+    ],
 
-    .. code-block:: py
+`replace`
+  Replace an asset file with one or multiple file(s).
 
-        'web.assets_common': [
-            ('replace', 'web/static/src/js/boot.js', 'my_addon/static/src/js/boot.js'),
-        ],
+  Let us say that an asset need not only to be removed, but you also want to insert
```
Comment (aab-odoo) [inline content/developer/reference/frontend/assets.rst:189]: need**s**

File: content/developer/reference/frontend/assets.rst [hunk 5]
```diff
@@ -38,191 +42,193 @@ a dedicated `assets` key which contains a dictionary. The dictionary will map
...
 The order in which assets are loaded is sometimes critical and must be deterministic,
 mostly for stylesheets priorities and setup scripts. Assets in Odoo are processed
 as follows.
 
-1. When an asset bundle is called (e.g. `t-call-assets="web.assets_common"`), an empty
+#) When an asset bundle is called (e.g. `t-call-assets="web.assets_common"`), an empty
 list of assets is generated
 
-2. All records of type ``ir.asset`` matching the bundle will be fetched and sorted
+#) All records of type ``ir.asset`` matching the bundle will be fetched and sorted
```
Comment (aab-odoo) [inline content/developer/reference/frontend/assets.rst:214]: will be -> are

Comment (aab-odoo) [reply content/developer/reference/frontend/assets.rst:214]: actually in this section, everything should be converted to present tense

File: content/developer/reference/frontend/assets.rst [hunk 6]
```diff
@@ -290,3 +299,48 @@ manifest counterparts.
...
+      * - `cssLibs`
+        - `string[]`
+        - a list of urls of css files
+
+
+.. js:function:: useAssets(assets)
+
+    :param Object assets: a description of various assets that should be loaded
+
+    This hook is useful when components needs to load some assets in their
```
Comment (aab-odoo) [inline content/developer/reference/frontend/assets.rst:345]: need

**Review by aab-odoo — APPROVED** [review]

File: content/developer/reference/frontend/assets.rst [hunk 7]
```diff
@@ -38,191 +42,195 @@ a dedicated `assets` key which contains a dictionary. The dictionary will map
...
 
-- `web.assets_frontend`: this bundle is about all that is specific to the public
-  website: ecommerce, forum, blog, event management, ...
+Typically, handling assets is simple: you just need to add some new files
+to a frequently used bundle like `common` or `backend`. But there are other operations
+available to cover some more specific use cases:
 
-Operations on asset bundles
----------------------------
+`append`
```
Comment (Feyensv) [inline content/developer/reference/frontend/assets.rst:107]: Do you mind declaring a specific sub-section for each operation ?
There are 7 operations and a quick access on the right side without needing to scroll could be useful :).

File: content/developer/reference/frontend/assets.rst [hunk 8]
```diff
@@ -38,191 +42,195 @@ a dedicated `assets` key which contains a dictionary. The dictionary will map
...
-- `web.assets_backend`: this bundle contains the code specific to the web client
-  (notably the web client/action manager/views) and all static XML templates used
-  in the backend environment
+Operations on asset bundles
+===========================
 
-- `web.assets_frontend`: this bundle is about all that is specific to the public
-  website: ecommerce, forum, blog, event management, ...
+Typically, handling assets is simple: you just need to add some new files
+to a frequently used bundle like `common` or `backend`. But there are other operations
```
Comment (Feyensv) [inline content/developer/reference/frontend/assets.rst:104]:
```suggestion
to a frequently used bundle like `assets_common` or `assets_backend`. But there are other operations
```

File: content/developer/reference/frontend/assets.rst [hunk 9]
```diff
@@ -38,191 +42,195 @@ a dedicated `assets` key which contains a dictionary. The dictionary will map
...
-------------
-When the Odoo server is started, it checks the timestamp of each file in a bundle
-and, if necessary, create/recreate the corresponding bundles.
+Template
+    Templates (static `xml` files) are handled in a different way: they are simply
+    read from the file system whenever they are needed, and concatenated.
 
-Here are some important bundles that most developers will need to know:
+    Whenever the browser loads odoo, it calls the `/web/webclient/qweb/` controller
+    to fetch the templates.
```
Comment (Feyensv) [inline content/developer/reference/frontend/assets.rst:92]: Reference the qweb documentation ?

**Review by Feyensv — APPROVED** [review]
Very interesting page :eyes:

Might be interesting to add some depths for easier access to specific information and eventually rethink the page structure now that it's more complete (cf some comments, I think it can still be improved but I don't have a perfect structure in mind atm, I'll think about it :D).

@robodoo delegate+

File: content/developer/reference/frontend/assets.rst [hunk 10]
```diff
@@ -9,18 +9,22 @@ Managing assets in Odoo is not as straightforward as it is in some other apps.
 One of the reasons is that we have a variety of situations where some, but not all
 of the assets are required. For example, the needs of the web client, the point of
 sale app, the website or even the mobile application are different. Also, some
-assets may be large, but are seldom needed. In that case, we sometimes want them
-to be loaded lazily.
+assets may be large, but are seldom needed: in that case we may want them
+to be loaded lazily (on demand).
```
[the remark this answers is not in the document]
Comment (ged-odoo) [reply content/developer/reference/frontend/assets.rst:13]: haha, yeah, the actual section was written after this sentence. forgot about this

File: content/developer/reference/frontend/assets.rst [hunk 11]
```diff
@@ -38,191 +42,195 @@ a dedicated `assets` key which contains a dictionary. The dictionary will map
...
---------------------
+Loading order
+=============
 
 The order in which assets are loaded is sometimes critical and must be deterministic,
 mostly for stylesheets priorities and setup scripts. Assets in Odoo are processed
 as follows.
 
-1. When an asset bundle is called (e.g. `t-call-assets="web.assets_common"`), an empty
+#) When an asset bundle is called (e.g. `t-call-assets="web.assets_common"`), an empty
```
[the remark this answers is not in the document]
Comment (ged-odoo) [reply content/developer/reference/frontend/assets.rst:213]: obviously... :) didn't even notice the difference

**Comment by ged-odoo** [comment]
@robodoo r+

**Review by AntoineVDV — COMMENTED** [review]
@ged-odoo I have a few comments regarding the structure of the page. I have the feeling that it could be improved by making the following changes:
1. Swap the "Bundles" and "Asset type" section
2. Rename the "Operations on assets bundle" section to "Operations" and make it a sub-section of the "Bundles" section.
3. Rename the "Lazy loading assets" section to "Lazy loading".
4. Fix ‘website’ -> \`website\` in the "Loading order" section.

**Comment by ged-odoo** [comment]
sounds reasonable @AntoineVDV

**Pull request closed without merging**